test/unittests/capi_test.cpp

```diff
@@ -12,3 +16,224 @@ TEST(capi, validate)
     wasm_prefix[7] = 1;
     EXPECT_FALSE(fizzy_validate(wasm_prefix, sizeof(wasm_prefix)));
 }
+
+TEST(capi, parse)
+{
+    uint8_t wasm_prefix[]{0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00};
+    auto module = fizzy_parse(wasm_prefix, sizeof(wasm_prefix));
+    EXPECT_NE(module, nullptr);
+    fizzy_free_module(module);
+    wasm_prefix[7] = 1;
+    EXPECT_EQ(fizzy_parse(wasm_prefix, sizeof(wasm_prefix)), nullptr);
+}
+
+TEST(capi, instantiate)
+{
+    uint8_t wasm_prefix[]{0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00};
+    auto module = fizzy_parse(wasm_prefix, sizeof(wasm_prefix));
+    EXPECT_NE(module, nullptr);
+
+    auto instance = fizzy_instantiate(module, nullptr, 0);
+    EXPECT_NE(instance, nullptr);
+
+    fizzy_free_instance(instance);
+}
+
+TEST(capi, instantiate_imported_function)
+{
+    /* wat2wasm
+      (func (import "mod1" "foo1") (result i32))
+    */
+    const auto wasm = from_hex("0061736d010000000105016000017f020d01046d6f643104666f6f310000");
+    auto module = fizzy_parse(wasm.data(), wasm.size());
+    EXPECT_NE(module, nullptr);
+
+    EXPECT_EQ(fizzy_instantiate(module, nullptr, 0), nullptr);
+
+    module = fizzy_parse(wasm.data(), wasm.size());
+    EXPECT_NE(module, nullptr);
+
+    fizzy_external_function host_funcs[] = {
```

Comment:
I think this would be more readable:
```suggestion
    FizzyExternalFunction host_funcs[] = {
```